### PR TITLE
Fixes for *BSDs

### PR DESCRIPTION
**Pull request by johalun**
Replace linux-only eventfd() with pipe() to enable use on *BSDs for alsa.
Add FreeBSD to supported OSes. 
Print warning to stdout if using null device.

Note: not tested on Linux.

### CODE REVIEW

File: src/null/mod.rs [hunk 1]
```diff
@@ -33,6 +41,7 @@ impl Iterator for EndpointsIterator {
 
 #[inline]
 pub fn get_default_endpoint() -> Option<Endpoint> {
+    println!("Unknown target system, default to null device.");
```
Comment (tomaka) [inline src/null/mod.rs]: Please remove this println, it's unrelated to the rest of the PR

File: src/null/mod.rs [hunk 2]
```diff
@@ -14,14 +15,21 @@ use UnknownTypeBuffer;
...
     #[inline]
     pub fn run(&self) { loop { /* TODO: don't spin */ } }
 }
 
-#[derive(Default)]
 pub struct EndpointsIterator;
 
+impl Default for EndpointsIterator {
+    fn default() -> Self {
+        println!("Unknown target system, default to null device.");
```
Comment (tomaka) [inline src/null/mod.rs]: Same

**Comment by johalun** [comment]
We can do a "alsa-pipe" especially for Unix but since Linux also has pipe() same as Unix I don't see any reason for double implementations. 

I think showing the user that null devices is being used is good. It took me a while to track down why it didn't work. Why it couldn't find any devices.. Turned out it was using null device since freebsd was unsupported.

**Comment by johalun** [comment]
Fixed.

**Comment by tomaka** [comment]
> I think showing the user that null devices is being used is good. It took me a while to track down why it didn't work. Why it couldn't find any devices.. Turned out it was using null device since freebsd was unsupported.

I agree but a println is not the right solution for this.

**Pull request merged**